src/cljs/ataru/virkailija/views/virkailija_readonly.cljs

```diff
@@ -0,0 +1,152 @@
+                                                                       scroll-to-anchor]]
+            [taoensso.timbre :refer-macros [spy debug]]
+            [ataru.feature-config :as fc]))
+
+(defn text [field-descriptor application lang]
+  [:div.application__form-field
+   [:label.application__form-field-label
+    (str (-> field-descriptor :label lang) (required-hint field-descriptor))]
+   [:div
+    (let [answer       ((answer-key field-descriptor) (:answers application))
+          values       (:value answer)
+          multi-value? (or (seq? values) (vector? values))
+          cannot-edit? (:cannot-edit answer)]
+      (cond
+        cannot-edit? [:p.application__form-field-not-edited (:not-edited (get-translations lang application-view-translations))]
+        multi-value? (into [:ul.application__form-field-list] (for [value values] [:li value]))
+        :else (textual-field-value field-descriptor application :lang lang)))]])
+
+(defn attachment [field-descriptor application lang]
+  (when (fc/feature-enabled? :attachment)
+    (let [answer-key (keyword (answer-key field-descriptor))
+          values     (get-in application [:answers answer-key :value])]
+      [:div.application__form-field
+       [:label.application__form-field-label
+        (str (-> field-descriptor :label lang) (required-hint field-descriptor))]
+       [:div
+        (map-indexed (fn attachment->link [idx {file-key :key filename :filename size :size}]
+                       (let [text          (str filename " (" (size-bytes->str size) ")")
+                             component-key (str "attachment-div-" idx)]
+                         [:div {:key component-key}
+                          [:a {:href (str "/lomake-editori/files/" file-key)}
+                           text]]))
+                     values)]])))
+
+(declare field)
+
+(defn child-fields [children application lang ui]
+  (for [child children
+        :when (get-in ui [(keyword (:id child)) :visible?] true)]
+    [field child application lang]))
+
+(defn wrapper [content application lang children]
+  (let [ui (subscribe [:state-query [:application :ui]])]
+    (fn [content application lang children]
+      [:div.application__wrapper-element.application__wrapper-element--border
+       [:div.application__wrapper-heading
+        [:h2 (-> content :label lang)]
+        [scroll-to-anchor content]]
+       (into [:div.application__wrapper-contents]
+         (child-fields children application lang @ui))])))
+
+(defn row-container [application lang children]
+  (let [ui (subscribe [:state-query [:application :ui]])]
+    (fn [application lang children]
+      (into [:div] (child-fields children application lang @ui)))))
+
+(defn- extract-values [children answers]
+  (let [child-answers  (->> (map answer-key children)
+                            (select-keys answers))
+        ; applicant side stores values as hashmaps
+        applicant-side (map (comp
```

Comment:
Can we write this as:

```clojure
(map (comp 
      (map :value) 
      :values 
      second))
```

In other words, lift the explicit mapping over `values` into a transducer.